pep-9999.rst

```diff
@@ -0,0 +1,245 @@
+specific affected version in question.
+
+The purpose of this field is to provide consumers with the information
+necessary to provide a consise upgrade path to mitigate a given vulnerability,
+without requiring API producers to provide the entire set of events that may be
+present in the OSV schema.
+
+An example for how to derive this from the OSV schema format follows:
+
+.. code-block:: python
+
+    fixed_in = [
+        event["fixed"]
+        for affected in schema["affected"]
+        if current_version in affected["versions"]
+        for range in affected["ranges"]
+        for event in range["events"]
+        if "fixed" in event
+    ]
+
+
+Versions listed in the ``fixed_in`` field MUST be present in the ``versions``
+key defined in :pep:`700` after normalizing versions with :pep:`440` (or string
+equivalence for legacy non-:pep:`440` versions). Additionally, versions listed
+in ``fixed_in`` MUST have at least one corresponding file in the ``files`` key
+defined in :pep:`691`.
+
+
+Additional file information
+---------------------------
+
+One new key is added to the ``files`` key:
+
+- ``vulnerability-ids``: This field is mandatory. If a given file has known
+  vulnerabilities associated with it, it MUST contain a list of vulnerability
+  IDs, all of which MUST correspond to keys in the ``vulnerabilties`` top level
+  field. If the file has no known vulnerabilities, this field should be an
+  empty list.
+
+Differences between the legacy JSON API
+=======================================
+
+The following are key differences between this specification and the `Legacy
+PyPI JSON API
+<https://warehouse.pypa.io/api-reference/json.html#known-vulnerabilities>`__.
+
+Removed fields
+--------------
+
+The ``link`` and ``source`` fields have been removed and are no longer present.
+
+
+Differences between the OSV schema
+==================================
+
+The following are key differences between this specification and the `Open
+Source Vulnerability Schema <https://ossf.github.io/osv-schema/>`__.
+
+Python-specific Prefix must be used
+-----------------------------------
+
+The OSV schema provides support for multiple vulnerability identifier prefixes,
+to provide support for multiple ecosystems. However, this PEP is only focused
+on the Python ecosystem, and thus requires that the primary vulnerability ID
+begin with the ``PYSEC-`` prefix. All other identifiers with non ``PYSEC-``
+prefixes should be included in the ``aliases`` field.
+
+SemVer should not be used
+-------------------------
+
+The OSV schema provides support for event types that specify `SemVer
+<https://semver.org/>`__ versions and version ranges. However, because the
+Python packaging ecosystem predates the creation of the SemVer specification,
+many non-SemVer versions exist and are used in practice. As a result, indexes
+should not use the ``SEMVER`` type, and should always use the ``ECOSYSTEM``
+type when specifying versions.
```

Comment:
Suggestion: consider spelling out one of the SemVers in this paragraph in full as Semantic Versioning. Also helps for when people search for it in full.